### ISSUE TITLE
make Desktop IO function async and pass filesystem exp test

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/application-services#3718]
This has been carved off from the big spike.  Andrei and I have got something that more or less works as of today, but seems like it's probably a no-go in it's current form because threadsafety issues:

[https://github.com/dmose/gecko/pull/8](https://github.com/dmose/gecko/pull/8)

there are various possible alternatives that have come up with conversations with platform folks; still need to sort through these.....

┆Issue is synchronized with this [Jira Spike](https://jira.mozilla.com/browse/SYNC-1898)
┆Epic: Desktop nimbus-sdk
┆Sprint: SYNC - end 2020-11-20

**Comment by data-sync-user** [comment]
➤ Daniel Mosedale commented:

This is basically working, it needs a couple of things cleaned up before we can declare victory.

**Comment by data-sync-user** [comment]
➤ Daniel Mosedale commented:

We're pretty much set here.  It's not mozilla-central quality just yet, but that's not part of this ticket.
